File: bindings/javascript/Polycode/CoreInput.js
require('Polycode/EventDispatcher')

function CoreInput() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.CoreInput()
	}
	Object.defineProperties(this, {
		'simulateTouchWithMouse': { enumerable: true, configurable: true, get: CoreInput.prototype.__get_simulateTouchWithMouse, set: CoreInput.prototype.__set_simulateTouchWithMouse},
		'simulateTouchAsPen': { enumerable: true, configurable: true, get: CoreInput.prototype.__get_simulateTouchAsPen, set: CoreInput.prototype.__set_simulateTouchAsPen},
		'simulateMouseWithTouch': { enumerable: true, configurable: true, get: CoreInput.prototype.__get_simulateMouseWithTouch, set: CoreInput.prototype.__set_simulateMouseWithTouch},
		'keyRepeat': { enumerable: true, configurable: true, get: CoreInput.prototype.__get_keyRepeat, set: CoreInput.prototype.__set_keyRepeat},
		'mousePosition': { enumerable: true, configurable: true, get: CoreInput.prototype.__get_mousePosition, set: CoreInput.prototype.__set_mousePosition},
		'deltaMousePosition': { enumerable: true, configurable: true, get: CoreInput.prototype.__get_deltaMousePosition, set: CoreInput.prototype.__set_deltaMousePosition}
	})
}

CoreInput.MOUSE_BUTTON1 = 0
CoreInput.MOUSE_BUTTON2 = 1
CoreInput.MOUSE_BUTTON3 = 2

CoreInput.prototype = Object.create(EventDispatcher.prototype)

CoreInput.prototype.__get_simulateTouchWithMouse = function() {
	return Polycode.CoreInput__get_simulateTouchWithMouse(this.__ptr)
}

CoreInput.prototype.__set_simulateTouchWithMouse = function(val) {
	Polycode.CoreInput__set_simulateTouchWithMouse(this.__ptr, val)
}

CoreInput.prototype.__get_simulateTouchAsPen = function() {
	return Polycode.CoreInput__get_simulateTouchAsPen(this.__ptr)
}

CoreInput.prototype.__set_simulateTouchAsPen = function(val) {
	Polycode.CoreInput__set_simulateTouchAsPen(this.__ptr, val)
}

CoreInput.prototype.__get_simulateMouseWithTouch = function() {
	return Polycode.CoreInput__get_simulateMouseWithTouch(this.__ptr)
}

CoreInput.prototype.__set_simulateMouseWithTouch = function(val) {
	Polycode.CoreInput__set_simulateMouseWithTouch(this.__ptr, val)
}

CoreInput.prototype.__get_keyRepeat = function() {
	return Polycode.CoreInput__get_keyRepeat(this.__ptr)
}

CoreInput.prototype.__set_keyRepeat = function(val) {
	Polycode.CoreInput__set_keyRepeat(this.__ptr, val)
}

CoreInput.prototype.__get_mousePosition = function() {
	var retVal = new Vector2("__skip_ptr__")
	retVal.__ptr = 	Polycode.CoreInput__get_mousePosition(this.__ptr)
	return retVal
}

CoreInput.prototype.__set_mousePosition = function(val) {
	Polycode.CoreInput__set_mousePosition(this.__ptr, val.__ptr)
}

CoreInput.prototype.__get_deltaMousePosition = function() {
	var retVal = new Vector2("__skip_ptr__")
	retVal.__ptr = 	Polycode.CoreInput__get_deltaMousePosition(this.__ptr)
	return retVal
}

CoreInput.prototype.__set_deltaMousePosition = function(val) {
	Polycode.CoreInput__set_deltaMousePosition(this.__ptr, val.__ptr)
}

Duktape.fin(CoreInput.prototype, function (x) {
	if (x === CoreInput.prototype) {
		return;
	}
	Polycode.CoreInput__delete(x.__ptr)
})

CoreInput.prototype.getMousePosition = function() {
	var retVal = new Vector2("__skip_ptr__")
	retVal.__ptr = Polycode.CoreInput_getMousePosition(this.__ptr)
	return retVal
}

CoreInput.prototype.getKeyState = function(keyCode) {
	return Polycode.CoreInput_getKeyState(this.__ptr, keyCode)
}

CoreInput.prototype.getJoystickButtonState = function(joystickIndex,button) {
	return Polycode.CoreInput_getJoystickButtonState(this.__ptr, joystickIndex, button)
}

CoreInput.prototype.getJoystickAxisValue = function(joystickIndex,axis) {
	return Polycode.CoreInput_getJoystickAxisValue(this.__ptr, joystickIndex, axis)
}

CoreInput.prototype.getMouseDelta = function() {
	var retVal = new Vector2("__skip_ptr__")
	retVal.__ptr = Polycode.CoreInput_getMouseDelta(this.__ptr)
	return retVal
}

CoreInput.prototype.getMouseButtonState = function(mouseButton) {
	return Polycode.CoreInput_getMouseButtonState(this.__ptr, mouseButton)
}

CoreInput.prototype.getNumJoysticks = function() {
	return Polycode.CoreInput_getNumJoysticks(this.__ptr)
}

CoreInput.prototype.addJoystick = function(deviceID) {
	Polycode.CoreInput_addJoystick(this.__ptr, deviceID)
}

CoreInput.prototype.removeJoystick = function(deviceID) {
	Polycode.CoreInput_removeJoystick(this.__ptr, deviceID)
}

CoreInput.prototype.joystickAxisMoved = function(axisID,value,deviceID) {
	Polycode.CoreInput_joystickAxisMoved(this.__ptr, axisID, value, deviceID)
}

CoreInput.prototype.joystickButtonDown = function(buttonID,deviceID) {
	Polycode.CoreInput_joystickButtonDown(this.__ptr, buttonID, deviceID)
}

CoreInput.prototype.joystickButtonUp = function(buttonID,deviceID) {
	Polycode.CoreInput_joystickButtonUp(this.__ptr, buttonID, deviceID)
}

CoreInput.prototype.mouseWheelUp = function(ticks) {
	Polycode.CoreInput_mouseWheelUp(this.__ptr, ticks)
}

CoreInput.prototype.mouseWheelDown = function(ticks) {
	Polycode.CoreInput_mouseWheelDown(this.__ptr, ticks)
}

CoreInput.prototype.setMouseButtonState = function(mouseButton,state,ticks) {
	Polycode.CoreInput_setMouseButtonState(this.__ptr, mouseButton, state, ticks)
}

CoreInput.prototype.setMousePosition = function(x,y,ticks) {
	Polycode.CoreInput_setMousePosition(this.__ptr, x, y, ticks)
}

CoreInput.prototype.setKeyState = function(keyCode,newState,ticks) {
	Polycode.CoreInput_setKeyState(this.__ptr, keyCode, newState, ticks)
}

CoreInput.prototype.setDeltaPosition = function(x,y) {
	Polycode.CoreInput_setDeltaPosition(this.__ptr, x, y)
}

CoreInput.prototype.touchesBegan = function(touch,touches,ticks) {
	Polycode.CoreInput_touchesBegan(this.__ptr, touch, touches, ticks)
}

CoreInput.prototype.touchesMoved = function(touch,touches,ticks) {
	Polycode.CoreInput_touchesMoved(this.__ptr, touch, touches, ticks)
}

CoreInput.prototype.touchesEnded = function(touch,touches,ticks) {
	Polycode.CoreInput_touchesEnded(this.__ptr, touch, touches, ticks)
}

CoreInput.prototype.textInput = function(text) {
	Polycode.CoreInput_textInput(this.__ptr, text)
}

CoreInput.prototype.clearInput = function() {
	Polycode.CoreInput_clearInput(this.__ptr)
}

File: bindings/lua/Polycode/CoreInput.lua
require "Polycode/EventDispatcher"

class "CoreInput" (EventDispatcher)

CoreInput.MOUSE_BUTTON1 = 0
CoreInput.MOUSE_BUTTON2 = 1
CoreInput.MOUSE_BUTTON3 = 2

function CoreInput:__getvar(name)
	if name == "simulateTouchWithMouse" then
		return Polycode.CoreInput_get_simulateTouchWithMouse(self.__ptr)
	elseif name == "simulateTouchAsPen" then
		return Polycode.CoreInput_get_simulateTouchAsPen(self.__ptr)
	elseif name == "simulateMouseWithTouch" then
		return Polycode.CoreInput_get_simulateMouseWithTouch(self.__ptr)
	elseif name == "ignoreOffScreenTouch" then
		return Polycode.CoreInput_get_ignoreOffScreenTouch(self.__ptr)
	elseif name == "keyRepeat" then
		return Polycode.CoreInput_get_keyRepeat(self.__ptr)
	elseif name == "mousePosition" then
		local retVal = Polycode.CoreInput_get_mousePosition(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector2"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "deltaMousePosition" then
		local retVal = Polycode.CoreInput_get_deltaMousePosition(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector2"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
	if EventDispatcher["__getvar"] ~= nil then
		return EventDispatcher.__getvar(self, name)
	end
end

function CoreInput:__setvar(name,value)
	if name == "simulateTouchWithMouse" then
		Polycode.CoreInput_set_simulateTouchWithMouse(self.__ptr, value)
		return true
	elseif name == "simulateTouchAsPen" then
		Polycode.CoreInput_set_simulateTouchAsPen(self.__ptr, value)
		return true
	elseif name == "simulateMouseWithTouch" then
		Polycode.CoreInput_set_simulateMouseWithTouch(self.__ptr, value)
		return true
	elseif name == "ignoreOffScreenTouch" then
		Polycode.CoreInput_set_ignoreOffScreenTouch(self.__ptr, value)
		return true
	elseif name == "keyRepeat" then
		Polycode.CoreInput_set_keyRepeat(self.__ptr, value)
		return true
	elseif name == "mousePosition" then
		Polycode.CoreInput_set_mousePosition(self.__ptr, value.__ptr)
		return true
	elseif name == "deltaMousePosition" then
		Polycode.CoreInput_set_deltaMousePosition(self.__ptr, value.__ptr)
		return true
	end
	if EventDispatcher["__setvar"] ~= nil then
		return EventDispatcher.__setvar(self, name, value)
	else
		return false
	end
end
function CoreInput:CoreInput(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "EventDispatcher" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.CoreInput(unpack(arg))
	end
end

function CoreInput:getMousePosition()
	local retVal =  Polycode.CoreInput_getMousePosition(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector2"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function CoreInput:getKeyState(keyCode)
	local retVal = Polycode.CoreInput_getKeyState(self.__ptr, keyCode)
	return retVal
end

function CoreInput:getJoystickButtonState(joystickIndex, button)
	local retVal = Polycode.CoreInput_getJoystickButtonState(self.__ptr, joystickIndex, button)
	return retVal
end

function CoreInput:getJoystickAxisValue(joystickIndex, axis)
	local retVal = Polycode.CoreInput_getJoystickAxisValue(self.__ptr, joystickIndex, axis)
	return retVal
end

function CoreInput:getMouseDelta()
	local retVal =  Polycode.CoreInput_getMouseDelta(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector2"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function CoreInput:getMouseButtonState(mouseButton)
	local retVal = Polycode.CoreInput_getMouseButtonState(self.__ptr, mouseButton)
	return retVal
end

function CoreInput:getNumJoysticks()
	local retVal =  Polycode.CoreInput_getNumJoysticks(self.__ptr)
	return retVal
end

function CoreInput:addJoystick(deviceID)
	local retVal = Polycode.CoreInput_addJoystick(self.__ptr, deviceID)
end

function CoreInput:removeJoystick(deviceID)
	local retVal = Polycode.CoreInput_removeJoystick(self.__ptr, deviceID)
end

function CoreInput:joystickAxisMoved(axisID, value, deviceID)
	local retVal = Polycode.CoreInput_joystickAxisMoved(self.__ptr, axisID, value, deviceID)
end

function CoreInput:joystickButtonDown(buttonID, deviceID)
	local retVal = Polycode.CoreInput_joystickButtonDown(self.__ptr, buttonID, deviceID)
end

function CoreInput:joystickButtonUp(buttonID, deviceID)
	local retVal = Polycode.CoreInput_joystickButtonUp(self.__ptr, buttonID, deviceID)
end

function CoreInput:mouseWheelUp(ticks)
	local retVal = Polycode.CoreInput_mouseWheelUp(self.__ptr, ticks)
end

function CoreInput:mouseWheelDown(ticks)
	local retVal = Polycode.CoreInput_mouseWheelDown(self.__ptr, ticks)
end

function CoreInput:setMouseButtonState(mouseButton, state, ticks)
	local retVal = Polycode.CoreInput_setMouseButtonState(self.__ptr, mouseButton, state, ticks)
end

function CoreInput:setMousePosition(x, y, ticks)
	local retVal = Polycode.CoreInput_setMousePosition(self.__ptr, x, y, ticks)
end

function CoreInput:setKeyState(keyCode, newState, ticks)
	local retVal = Polycode.CoreInput_setKeyState(self.__ptr, keyCode, newState, ticks)
end

function CoreInput:setDeltaPosition(x, y)
	local retVal = Polycode.CoreInput_setDeltaPosition(self.__ptr, x, y)
end

function CoreInput:touchesBegan(touch, touches, ticks)
	local retVal = Polycode.CoreInput_touchesBegan(self.__ptr, touch.__ptr, touches.__ptr, ticks)
end

function CoreInput:touchesMoved(touch, touches, ticks)
	local retVal = Polycode.CoreInput_touchesMoved(self.__ptr, touch.__ptr, touches.__ptr, ticks)
end

function CoreInput:touchesEnded(touch, touches, ticks)
	local retVal = Polycode.CoreInput_touchesEnded(self.__ptr, touch.__ptr, touches.__ptr, ticks)
end

function CoreInput:textInput(text)
	local retVal = Polycode.CoreInput_textInput(self.__ptr, text)
end

function CoreInput:clearInput()
	local retVal =  Polycode.CoreInput_clearInput(self.__ptr)
end

function CoreInput:__delete()
	if self then Polycode.delete_CoreInput(self.__ptr) end
end
